src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,49 @@
+package duke;
+
+import duke.exception.InvalidDeadlineException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * A subclass of task that represents a deadline.
+ *
+ */
+public class Deadline extends Task {
+    private String type;
+    private LocalDate deadline;
+
+    /**
+     * Constructor for Deadline class.
+     *
+     * @param title Name of the task to be created.
+     * @param deadline Date of the deadline.
+     */
+    public Deadline(String title, String deadline) throws InvalidDeadlineException {
+        super(title);
+        this.type = "D";
```

Comment:
Perhaps you may not need the 'this' keyword here, since you are not assigning the instance variable 'type' to a parameter variable of the same name.

```suggestion
        type = "D";
```